core/src/main/scala/com/spotify/featran/transformers/Transformer.scala

```diff
@@ -70,6 +70,31 @@ abstract class Transformer[-A, B, C](val name: String) extends Serializable {
   protected def nameAt(n: Int): String = name + '_' + n
   protected def names(n: Int): Stream[String] = (0 until n).toStream.map(nameAt)
 
+  /**
+   * Builds a new transformer with an extra input preprocessing step
+   * @param f input preprocessing function
+   */
+  def contramap[AA](f: AA => A): Transformer[AA, B, C] = {
+    val self = this
+    new Transformer[AA, B, C](name) {
+      val aggregator: Aggregator[AA, B, C] = new Aggregator[AA, B, C] {
+        override def prepare(a: AA): B = self.aggregator.prepare(f(a))
+        override def semigroup = self.aggregator.semigroup
+        override def present(b: B): C = self.aggregator.present(b)
+      }
+      def buildFeatures(a: Option[AA], c: C, fb: FeatureBuilder[_]): Unit =
+        self.buildFeatures(a.map(f), c, fb)
+      def decodeAggregator(s: String): C = self.decodeAggregator(s)
+      def encodeAggregator(c: C): String = self.encodeAggregator(c)
+      def featureDimension(c: C): Int = self.featureDimension(c)
+      def featureNames(c: C): Seq[String] = self.featureNames(c)
+      def flatRead[T](implicit evidence$1: FlatReader[T]): T => Option[Any] =
```

Comment:
🎨 we can just use `FlatReader` as a context bound

Reply:
Addressed in 014c640